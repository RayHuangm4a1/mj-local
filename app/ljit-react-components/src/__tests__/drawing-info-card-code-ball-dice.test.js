import React from 'react';
import { shallow, mount, } from 'enzyme';

describe('DrawingInfoCardCodeBallDice', () => {
	let DrawingInfoCardCodeBallDice;

	beforeEach(() => {
		DrawingInfoCardCodeBallDice = require('../components/drawing-info-card/elements/drawing-info-card-code-ball-dice').default;
	});

	afterEach(() => {
		DrawingInfoCardCodeBallDice = undefined;
	});

	it('should be selectable by class drawing-info-card-code-ball-dice', () => {
		const wrapper = shallow(<DrawingInfoCardCodeBallDice
			splitOpencodes={['1','2','3','4','5',]}
		/>);

		expect(wrapper.hasClass('drawing-info-card-code-ball-dice')).toEqual(true);
	});

	it('should renders correctly', () => {
		const props = {
			splitOpencodes: ['1','2','3','4','5',],
		};

		const wrapper = shallow(<DrawingInfoCardCodeBallDice {...props} />);

		expect(wrapper).toMatchSnapshot();
	});

	it('should mount in a full DOM', () => {
		const props = {
			splitOpencodes: ['1','2','3','4','5',],
		};

		const wrapper = mount(<DrawingInfoCardCodeBallDice {...props} />);

		expect(wrapper.props().opencode).toBe(props.opencode);
		expect(wrapper.props().splitOpencode).toBe(props.splitOpencode);
	});
});
