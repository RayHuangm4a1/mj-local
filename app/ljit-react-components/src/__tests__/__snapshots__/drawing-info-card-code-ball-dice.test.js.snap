// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawingInfoCardCodeBallDice should renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DrawingInfoCardCodeBallDice
    splitOpencodes={
      Array [
        "1",
        "2",
        "3",
        "4",
        "5",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "className": "drawing-info-card-code-ball-dice",
      "points": Array [
        1,
        2,
        3,
        4,
        5,
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "className": "drawing-info-card-code-ball-dice",
        "points": Array [
          1,
          2,
          3,
          4,
          5,
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
